lib/internal/Magento/Framework/View/Layout/GeneratorPool.php

```diff
@@ -69,14 +72,17 @@ public function getGenerator($type)
     /**
      * Traverse through all generators and generate all scheduled elements
      *
-     * @param Reader\Context $readerContext
+     * @param Reader\Context    $readerContext
      * @param Generator\Context $generatorContext
+     *
      * @return $this
+     * @throws LocalizedException
      */
     public function process(Reader\Context $readerContext, Generator\Context $generatorContext)
     {
         $this->buildStructure($readerContext->getScheduledStructure(), $generatorContext->getStructure());
         foreach ($this->generators as $generator) {
+            $this->validateGenerator($generator);
```

Comment:
Please validate in `addGenerators` only and get rid of this method.

Reply:
@orlangur O believe this validation should be there too since the argument `$generators` can be injected via DI as well as using `addGenerator` method. Please let me know what you think about it.

Reply:
How? As I see https://github.com/magento/magento2/blob/9c4e1758640ef6be4458929fc61d2619a78d9fc4/lib/internal/Magento/Framework/View/Layout/GeneratorPool.php#L51 it is still added via `addGenerators` this way.

Reply:
Forget what I've said. It really doesn't make sense and I just made the commit.